### PR TITLE
Quiz editor

### PR DESCRIPTION
**Pull request by jasminetai**
When you click the "Create quiz" button on the upload page, you get sent to a quiz editor page, where you can create a quiz to either download or begin hosting immediately. There's a Kakaw bird in the corner to assist you with setting the quiz title, default timer, and questions/answers. This works nicely... I think.

Notes:
- for now, the author is simply "Host", and points are always 1000
- answers can be added only so long as there is a color for it in QuestionEditor.tsx, so that's 4 answers (later on, we can finalize our colors (six?) and revise this easily)
- new questions always get assigned initial times equal to the current quiz default time
- no support for deleting questions or answers yet
- warning message for if a quiz is invalid and you try to start is generic, a slight deviation from the Figma design since it errors if anything is wrong with the quiz (rather than only if there are no correct answers)
- downloading a quiz does not have any checks, so it is possible to download quizzes that are incomplete

Probably don't merge until the connection page routing PR goes through, as I merged from that branch recently so that I could see the quiz start working for myself.

### CODE REVIEW

File: frontend/public/fail_kaw.png:Zone.Identifier
Comment (celdond) [inline frontend/public/fail_kaw.png:Zone.Identifier:1]: From what I know, zone identifiers are not supposed to be committed.  The editor works without them (my computer auto deletes them and will not let me restore them).

**Review by celdond — DISMISSED** [review]
Great work, quiz editor looks great!
A bit of a question, and I marked it, I have never dealt with data streams before, but, from what I have read, Zone Identifiers are actually not supposed to be committed.  I only mention this because my computer refuses to allow them on my system, and it barred me from even switching branches until I used a work around.

File: frontend/public/fail_kaw.png:Zone.Identifier
Comment (jasminetai) [reply frontend/public/fail_kaw.png:Zone.Identifier:1]: You're completely right, that shouldn't be there. Thanks for catching it!

**Comment by 190n** [comment]
This looks great.

> answers can be added only so long as there is a color for it in QuestionEditor.tsx, so that's 4 answers (later on, we can finalize our colors (six?) and revise this easily)

@Shea-Scarlet mentioned purple and pink on Discord so I wrote that into `QuestionAnswers.tsx`.

> warning message for if a quiz is invalid and you try to start is generic, a slight deviation from the Figma design since it errors if anything is wrong with the quiz (rather than only if there are no correct answers)

Could you also mention the requirement for at least 2 answers per question in this message?

**Comment by jasminetai** [comment]
For the clarification on 2 answers per question, I can certainly do that. For the colors, I see that now, but it seems like you use bg-red-400, bg-blue-400, etc for the four original colors, while I have used the custom colors from Figma, defined as bg-red-200, bg-blue-200, etc. Which colors would we like to go with?

**Comment by 190n** [comment]
Custom colors seem better if those are ones Gea chose.

**Comment by 190n** [comment]
Should purple and pink be custom `-200` colors as well? To be clear I'm talking about the colors [here](https://canary.discord.com/channels/1095163007199354990/1098435056185385031/1109276582180307005) -- #b497ff (purple) and #e086ff (pink). I wasn't very careful with how I integrated those on my branch, I was mostly focused on logic changes.

**Comment by jasminetai** [comment]
I’m not sure that Gea chose specific colors for pink and purple, so I didn’t add those for now.

File: frontend/public/fail_kaw.png:Zone.Identifier
Comment (celdond) [reply frontend/public/fail_kaw.png:Zone.Identifier:1]: I saw you removed the other identifier, remember this one is here as well.

**Review by celdond — APPROVED** [review]
I have no more concerns, good work!

**Review by 190n — APPROVED** [review]

**Pull request merged**